show-changes

```diff
@@ -48,38 +48,66 @@ for section in security bugfixes changes updates; do
+      if [ "$(git -C scripts show "${OLD}":.gitmodules 2>/dev/null)" != "" ]; then
+        # Find the pinned submodule refs because there may be no release tags inside the submodules
+        # Pipe to awk instead of using --object-only for git 2.35 support
+        OLDREF=$(git -C "scripts" ls-tree "${OLD}" "sdk_container/src/third_party/${repo}" | awk '{print $3 }')
+        # We can't assume anymore that the submodule repo is available under scripts/
+        if [ ! -d "${repo}" ]; then
+          git clone "git@github.com:flatcar/${repo}.git"
+        fi
+      else
+        OLDPREPEND="sdk_container/src/third_party/${repo}/"
+        OLDREPOPATH="scripts"
+      fi
+      if [ "$(git -C scripts show "${NEW}":.gitmodules 2>/dev/null)" != "" ]; then
+        NEWREF=$(git -C "scripts" ls-tree "${NEW}" "sdk_container/src/third_party/${repo}" | awk '{print $3 }')
+        if [ ! -d "${repo}" ]; then
+          git clone "git@github.com:flatcar/${repo}.git"
+        fi
+      else
+        NEWPREPEND="sdk_container/src/third_party/${repo}/"
+        NEWREPOPATH="scripts"
+      fi
     fi
     if [ "${FETCH}" = 1 ]; then
-      git -C "${REPOPATH}" fetch -t -f 2> /dev/null > /dev/null || { echo "Error: git fetch -t -f failed" ; exit 1 ; }
+      git -C "${OLDREPOPATH}" fetch -t -f 2> /dev/null > /dev/null || { echo "Error: git fetch -t -f failed" ; exit 1 ; }
+      git -C "${NEWREPOPATH}" fetch -t -f 2> /dev/null > /dev/null || { echo "Error: git fetch -t -f failed" ; exit 1 ; }
     fi
     if [ "${section}" = "security" ] && [ "${repo}" = "coreos-overlay" ]; then
-      FROM_KERNEL=$(git -C "${REPOPATH}" show "${OLDREF}":sys-kernel/coreos-kernel/ | grep -m 1 'coreos-kernel-.*\.ebuild' | cut -d - -f 3 | cut -d . -f 1-3)
-      TO_KERNEL=$(git -C "${REPOPATH}" show "${NEWREF}":sys-kernel/coreos-kernel/ | grep -m 1 'coreos-kernel-.*\.ebuild' | cut -d - -f 3 | cut -d . -f 1-3)
+      FROM_KERNEL=$(git -C "${OLDREPOPATH}" show "${OLDREF}":"${OLDPREPEND}"sys-kernel/coreos-kernel/ | grep -m 1 'coreos-kernel-.*\.ebuild' | cut -d - -f 3 | cut -d . -f 1-3)
+      TO_KERNEL=$(git -C "${NEWREPOPATH}" show "${NEWREF}":"${NEWPREPEND}"sys-kernel/coreos-kernel/ | grep -m 1 'coreos-kernel-.*\.ebuild' | cut -d - -f 3 | cut -d . -f 1-3)
       if [ "${FROM_KERNEL}" != "${TO_KERNEL}" ]; then
         KERNEL_ENTRIES=$("${SCRIPTFOLDER}"/show-fixed-kernel-cves.py -f "${FROM_KERNEL}" -t "${TO_KERNEL}")
         if [ "${KERNEL_ENTRIES}" != "" ]; then
           echo "- Linux (${KERNEL_ENTRIES})"
         fi
       fi
     fi
-    git -C "${REPOPATH}" difftool --no-prompt --extcmd='sh -c "cat \"$REMOTE\"" --' "${OLDREF}..${NEWREF}" -- "changelog/${section}/" | sort || { echo "Error: git difftool failed" ; exit 1 ; }
+
+    # The assumption is that the old ref is really older, so we can assume that old would have submodules while new doesn't have them anymore
+    if [ "${OLDREPOPATH}" != "${NEWREPOPATH}" ] && [ "${NEWREPOPATH}" = "scripts" ]; then
```

Comment:
I'm wondering if this handling will actually be needed, because we plan the next releases to use the submodules for both old and new releases, then we switch to merged scripts by retagging the newest release. So after that we are going to have both old and new release using merged scripts.

Also, I'm thinking that with merged scripts repos, we will abandon the changelog directories in `sdk_container/src/third_party/<repo>` in favor of using the toplevel one, right?

Reply:
I think it's good to be able to diff releases instead of having a had break.

Yes, future entries can directly go to `/changelog/` in scripts but the old ones should stay there.

Reply:
Having this also is an alternative to doing the retag, which I think it more elegant - specially when there are follow-up fixes needed to get the build working and other changes also start to slip in.